src/box/lua/config/configdata.lua

```diff
@@ -147,6 +147,10 @@ function methods.sharding(self)
     for group_name, group in pairs(self._cconfig.groups) do
         for replicaset_name, value in pairs(group.replicasets) do
             local lock
+            local weight
+            -- For replicaset-level options, we need to get them from the first
+            -- instance of the replicaset.
+            local is_first = true
```

Comment:
Nit: It sounds for me like we're obligated to acquire the values from the first instance (not any other), while the idea here is that we can acquire it from any instance of the replicaset (so, the check is needed only to skip unnecessary actions).

It confused me on the first reading: I had thoughts how the first instance is different and what is the order in which we traverse them here.

Reply:
Replaced by `is_rs_options_set`. Also replaced `first` in the comment by `any`.